account_invoice_import_invoice2data/wizard/account_invoice_import.py

```diff
@@ -90,9 +90,74 @@ def invoice2data_to_parsed_inv(self, invoice2data_res):
             'date_start': invoice2data_res.get('date_start'),
             'date_end': invoice2data_res.get('date_end'),
             'description': invoice2data_res.get('description'),
+            'mapping_lines': invoice2data_res.get('lines', [])
             }
         if 'amount_untaxed' in invoice2data_res:
             parsed_inv['amount_untaxed'] = invoice2data_res['amount_untaxed']
         if 'amount_tax' in invoice2data_res:
             parsed_inv['amount_tax'] = invoice2data_res['amount_tax']
         return parsed_inv
+
+    @api.model
+    def _prepare_create_invoice_vals(self, parsed_inv):
+        """ Extend invoice with line mapping. """
+        vals = super(AccountInvoiceImport, self)._prepare_create_invoice_vals(
+            parsed_inv)
+        template = self.env['invoice2data.template'].search([
+            ('related_partner.name', '=', parsed_inv.get('partner')['name'])
+        ])
+        if parsed_inv.get('mapping_lines'):
```

Comment:
I don't see where you check that the config is set to `product_mapping`. Should you do that here?

Reply:
Indeed, i missed the validation. Added!